### ISSUE TITLE
Is it possible to disable the signup option?

### ISSUE THREAD

**Issue by thomasjsn** [xwiki-labs/cryptpad#242]
I'm self-hosting an instance of Cryptpad, and after creating users for the family members I'd like to disable signup. Can't find any option for that in the `config.js` file, is it possible?

**Comment by ansuz** [comment]
At the moment we don't have a simple configuration point for disabling registration, however, you can customize your CryptPad instance by creating a `/customize/` directory. Any of the files in `/customize.dist/` can be overridden by creating a file in `/customize/` with the same name.

`/customize.dist/` contains [login.js](https://github.com/xwiki-labs/cryptpad/blob/master/customize.dist/login.js), which implements much of our login and registration logic. Assuming you have some proficiency in Javascript, you should be able to stub some of the logic there to prevent new registrations.

**Comment by casipw** [comment]
I had [the same requirement](https://github.com/xwiki-labs/cryptpad/issues/285) and modified the following files in `./customize`:

* `login.js`: truncated
* `pages.js`, `template.js`: removed all code around `login` and `register`
* `*.html`: applied a patch (`html-patch`, attached) which hides the "Log in" and "Sign up" buttons

Here's my `customize` folder:

[cryptpad-customize.zip](https://github.com/xwiki-labs/cryptpad/files/2472869/cryptpad-customize.zip)

**Comment by ansuz** [comment]
Hi all,

It's encouraging to see people modifying their instances to suit their needs. Unfortunately, modifying html and javascript files in `/customize/`can be a little fragile, depending on the file. Updating your CryptPad instance might break certain features, and we don't backport bugfixes (including security vulnerabilities), so you may find yourself unable to update without taking time to modify your customizations.

We've been discussing how to make modifications easier, but there's some work involved in updating our architecture. While we're going to go ahead with these improvements at some point, we would certainly appreciate contributions to our [OpenCollective funding campaign](https://opencollective.com/cryptpad). If these improvements are especially important for you, opencollective donations can be accompanied by a message indicating what features you'd like your contributions to fund.

**Comment by pintman** [comment]
Some kind of workaround might be to cut down storage limits in general (in config.js) and give each registered user a custom limit. This certainly only works for a small number of users.

**Comment by Leopere** [comment]
I would love to avoid competing with the paid model of subscription available here with Cryptpad however I don't want to have to hack my way through javascript to do it.  This should be a priority here because of this and #358 should be a potential option because I only need this within my org and I'm already paying for the server so I would prefer to avoid giving accounts to anyone who just shows up at my domain.

**Comment by targodan** [comment]
I would also like to be able to deactivate registration. I only host the instance for a small team and I don't want my instance to be "public".

Either deactivating registration all together via a flag would be nice. Or even something like shared-secret registration or invite only.

**Comment by ansuz** [comment]
The ability to restrict registration to users redeeming an admin-generated invitation link is on our roadmap (contingent on being awarded a grant for which we have submitted a proposal).

**Comment by AndrewPaglusch** [comment]
I know this is far from an ideal solution, but this is how I've disabled registration until this feature is implemented.

Nginx Config:
```
location /register {
    deny all;
}
```

**Comment by sbernhard** [comment]
It would be great to run cryptpad in a mode without register / login functionality. Normally, I would prefer to have openid / saml authentication but it looks like its completely out of scope (https://github.com/xwiki-labs/cryptpad/issues/174). Therefore, it should be possible to work with cryptpad in a "anonymous" mode:
- user access cryptpad
- need to enter a name (which will be used if you work together in a group on one document)
- everything can be done - even uploading a file

**Comment by Silur** [comment]
seems like there is a flag for this in the admin panel but it's disabled: https://github.com/xwiki-labs/cryptpad/blob/main/www/admin/inner.js#L50

**Comment by matbgn** [comment]
@ansuz You mentioned a donation "way to solve" this problem

> We've been discussing how to make modifications easier, but there's some work involved in updating our architecture. While we're going to go ahead with these improvements at some point, we would certainly appreciate contributions to our [OpenCollective funding campaign](https://opencollective.com/cryptpad). If these improvements are especially important for you, opencollective donations can be accompanied by a message indicating what features you'd like your contributions to fund.

What do you mean in regards of amount, that would make this feature on the road for future releases? Quiet difficult to invest money for waiting two years. Of course you deserve it, but I'm not Cresus unfortunately :cry:

**Comment by ansuz** [comment]
Registration can be disabled via the admin panel as of [4.5.0](https://github.com/xwiki-labs/cryptpad/releases/tag/4.5.0).

**Comment by ansuz** [comment]
This feature was funded by NLnet foundation as a part of our [CryptPad for Communities](https://nlnet.nl/project/Cryptpad-Communities/) R&D project.

**Comment by KiaraGrouwstra** [comment]
a roundabout way of achieving this seems to be to use SSO, see #749
